### DISCUSSION TITLE
Return type of useDebouncedCallback is not derived from the passed-in function

### DISCUSSION THREAD

**Discussion by moonray** [mantinedev/mantine#6986] [Feature requests]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.13.2

### What package has an issue?

@mantine/hooks

### What framework do you use?

Other, I will specify in the bug description

### In which browsers you can reproduce the issue?

None

### Describe the bug

The return type for useDebouncedCallback is `void` instead of matching the return type of the function that was passed in.
So if the callback function returns something other than undefined/void, the return type is incorrect.

EDIT:
Looking more closely at the useDebouncedCallback function it seems it's not built to actually have the debounced function return what the callback function returns. So that would be a good thing to implement. A good example would be the debounce-promise package.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by jessepinho-oai** [comment]
Agreed. I expected `useDebouncedCallback()` to have the same behavior as `useCallback` — i.e., it passes along the return value.
